### PR TITLE
Detect dependencies by listing MANIFEST.MF files at runtime

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

This allows us to collect dependencies at runtime by looking at all available `MANIFEST.MF` resources and extracting the dependency name and version from the URL. This does not actually look at the contents of the manifest files but only at the URLs which usually takes a low single digit amount of `ms` to collect all dependencies.

## :bulb: Motivation and Context

Allows us to show `modules` in UI for JVM applications that aren't using our plugin to provide `sentry-external-modules.txt`. A workaround for #2474 until we can provide a non Android focused build plugin that users can use.

## :green_heart: How did you test it?
Running Spring Boot from IDE, from runnable JAR and WAR deployed to Tomcat. Unit Tests

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [-] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry/src/main/java/io/sentry/Sentry.java [hunk 1]
```diff
@@ -276,10 +280,18 @@ private static boolean initConfigurations(final @NotNull SentryOptions options)
               });
     }
 
-    final IModulesLoader modulesLoader = options.getModulesLoader();
-    // only override the ModulesLoader if it's not already set by Android
+    final @NotNull IModulesLoader modulesLoader = options.getModulesLoader();
     if (modulesLoader instanceof NoOpModulesLoader) {
-      options.setModulesLoader(new ResourcesModulesLoader(options.getLogger()));
+      if (Platform.isJvm()) {
```
Comment (romtsn) [inline sentry/src/main/java/io/sentry/Sentry.java:285]: `m`: I think we don't need this check, 'cause this was the codepath for JVM platforms anyway, so I think CompositeModulesLoader covers everything already

File: sentry/src/main/java/io/sentry/internal/modules/ManifestModulesLoader.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          classLoader.getResources("META-INF/MANIFEST.MF");
+      while (manifestUrls.hasMoreElements()) {
+        final @NotNull URL manifestUrl = manifestUrls.nextElement();
+        final @Nullable String originalName = extractDependencyNameFromUrl(manifestUrl);
+        final @Nullable Module module = convertOriginalNameToModule(originalName);
+        if (module != null) {
+          modules.add(module);
+        }
+      }
+    } catch (IOException e) {
```
Comment (romtsn) [inline sentry/src/main/java/io/sentry/internal/modules/ManifestModulesLoader.java:61]:
```suggestion
    } catch (Thowable e) {
```
Just to be on the safe side

File: sentry/src/main/java/io/sentry/internal/modules/ManifestModulesLoader.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+  ManifestModulesLoader(final @Nullable ClassLoader classLoader, final @NotNull ILogger logger) {
+    super(logger);
+    this.classLoader = classLoaderOrDefault(classLoader);
+  }
+
+  @Override
+  protected Map<String, String> loadModules() {
+    final @NotNull Map<String, String> modules = new HashMap<>();
+    List<Module> detectedModules = detectModulesViaManifestFiles();
```
Comment (romtsn) [inline sentry/src/main/java/io/sentry/internal/modules/ManifestModulesLoader.java:39]: `m`: one minor question - are the modules alphabetically sorted within the MANIFEST.MF file? If not, I guess we'd have to sort them, not sure if there's sorting of those somewhere on relay or frontend.

**Review by romtsn — APPROVED** [review]

File: sentry/src/main/java/io/sentry/Sentry.java (hunk 1 above)
Comment (adinauer) [reply sentry/src/main/java/io/sentry/Sentry.java:285]: OK Great, I presume this also won't cause problems with other SDKs using the Java SDK, correct?

File: sentry/src/main/java/io/sentry/internal/modules/ManifestModulesLoader.java (hunk 3 above)
Comment (adinauer) [reply sentry/src/main/java/io/sentry/internal/modules/ManifestModulesLoader.java:39]: They're not sorted when sending the JSON but show up sorted in the UI. Guess that's fine then?

**Pull request merged**